runtime/gc_vlhgc/ProjectedSurvivalCollectionSetDelegate.cpp

```diff
@@ -557,6 +561,39 @@ MM_ProjectedSurvivalCollectionSetDelegate::rateOfReturnCalculationBeforeSweep(MM
 				}
 			}
 		}
+#if defined(J9VM_GC_SPARSE_HEAP_ALLOCATION)
+		if (_extensions->isVirtualLargeObjectHeapEnabled) {
+			const uintptr_t arrayletLeafSize = env->getOmrVM()->_arrayletLeafSize;
+			MM_SparseVirtualMemory *largeObjectVirtualMemory = _extensions->largeObjectVirtualMemory;
+			uintptr_t arrayletLeafCount = 0;
+			J9HashTableState walkState;
+
+			MM_SparseDataTableEntry *sparseDataEntry = (MM_SparseDataTableEntry *)hashTableStartDo(largeObjectVirtualMemory->getSparseDataPool()->getObjectToSparseDataTable(), &walkState);
+			while (NULL != sparseDataEntry) {
+				J9Object *spineObject = (J9Object *)sparseDataEntry->_proxyObjPtr;
+				uintptr_t dataSize = sparseDataEntry->_size;
+				/* TODO: how fraction is counting here? */
+//				arrayletLeafCount = MM_Math::roundToFloor(arrayletLeafSize, dataSize) / arrayletLeafSize;
+				arrayletLeafCount = MM_Math::roundToCeiling(arrayletLeafSize, dataSize) / arrayletLeafSize;
```

Comment:
ceiling is more correct for now (remove the floor attempt) , but when we start sharing arraylet regions we will do floor and the after the loop we will add  +1 if we have a shared arraylet region (possible for each Allocation Context)